### ISSUE TITLE
NavigationMapRoute manually progress listener addition is ignored

### ISSUE THREAD

**Issue by LukasPaczos** [mapbox/mapbox-navigation-android#2728]
Looking at 
https://github.com/mapbox/mapbox-navigation-android/blob/ded442ede603287004b1dd442da499a33a01421c/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java#L304-L308

The listener should be registered instead of unregistered.

/cc @abhishek1508 @cafesilencio @JunDai

**Comment by JunDai** [comment]
hey @LukasPaczos , were you saying the `addProgressChangeListener` should do the **register**?
If so, we have fixed the typo here #2705 
please refer to the latest code [here](https://github.com/mapbox/mapbox-navigation-android/blob/master/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java#L292-L308)

```java
  public void addProgressChangeListener(MapboxNavigation navigation) {
    this.navigation = navigation;
    navigation.registerRouteProgressObserver(mapRouteProgressChangeListener);
  }


  /**
   * Should be called if {@link NavigationMapRoute#addProgressChangeListener(MapboxNavigation)} was
   * called to prevent leaking.
   *
   * @param navigation to remove the progress change listener
   */
  public void removeProgressChangeListener(MapboxNavigation navigation) {
    if (navigation != null) {
      navigation.unregisterRouteProgressObserver(mapRouteProgressChangeListener);
    }
  }

**Comment by LukasPaczos** [comment]
Awesome, thanks @JunDai! My GitHub page must've been checked out on a tag when I browsed the code.
